File: Java/src/DLinkedList/Node.java
package DLinkedList;

public class Node<T> {
    private T data;
    private Node<T> prev;
    private Node<T> next;

    public Node(Node<T> next, T data, Node<T> prev) {
        this.next = next;
        this.data = data;
        this.prev = prev;
    }

    ;

    public Node<T> getPrev() {
        return prev;
    }

    public void setPrev(Node<T> prev) {
        this.prev = prev;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    public Node<T> getNext() {
        return next;
    }

    public void setNext(Node<T> next) {
        this.next = next;
    }

    @Override
    public String toString() {
        return data.toString();
    }
}

File: Typescript/src/LinkedList/Node.ts
class NODE<T> {
    private data: T | null;
    private next: NODE<T> | null;
    private prev: NODE<T> | null;

    constructor(prev: NODE<T> | null, data: T, next: NODE<T> | null) {
        this.data = data;
        this.next = next;
        this.prev = prev;
    }

    public getData(): T | null {
        return this.data;
    }

    public setData(data: T | null): void {
        this.data = data;
    }

    public getNext(): NODE<T> | null {
        return this.next;
    }

    public setNext(next: NODE<T> | null): void {
        this.next = next;
    }

    public getPrev(): NODE<T> | null {
        return this.prev;
    }

    public setPrev(prev: NODE<T> | null): void {
        this.prev = prev;
    }
}

export default NODE;
